extensions/blocks/opentable/edit.js

```diff
@@ -0,0 +1,294 @@
+
+	return (
+		<form onSubmit={ () => onSubmit( embedCode ) }>
+			<TextareaControl
+				onChange={ value => setEmbedCode( value ) }
+				placeholder={ __( 'Paste your OpenTable embed code here…', 'jetpack' ) }
+			>
+				{ embedCode }
+			</TextareaControl>
+			<div>
+				<Button isLarge type="submit">
+					{ _x( 'Embed', 'button label', 'jetpack' ) }
+				</Button>
+			</div>
+		</form>
+	);
+};
+
+export default function OpenTableEdit( { attributes, setAttributes, className, clientId } ) {
+	const validatedAttributes = getValidatedAttributes( defaultAttributes, attributes );
+
+	if ( ! isEqual( validatedAttributes, attributes ) ) {
+		setAttributes( validatedAttributes );
+	}
+
+	const { rid, style, iframe, domain, lang, newtab } = attributes;
+	const [ notice, setNotice ] = useState();
+
+	const setErrorNotice = () =>
+		setNotice(
+			<>
+				<strong>{ __( 'We ran into an issue', 'jetpack' ) }</strong>
+				<br />
+				{ __( 'Please ensure this embed matches the one from your OpenTable account', 'jetpack' ) }
+			</>
+		);
+
+	const parseEmbedCode = embedCode => {
+		if ( ! embedCode ) {
+			setErrorNotice();
+			return;
+		}
+
+		const scriptTagAttributes = embedCode.match( /< *script[^>]*src *= *["']?([^"']*)/i );
+		if ( ! scriptTagAttributes || ! scriptTagAttributes[ 1 ] ) {
+			setErrorNotice();
+			return;
+		}
+
+		let src = '';
+		if ( scriptTagAttributes[ 1 ].indexOf( 'http' ) === 0 ) {
+			src = new URL( scriptTagAttributes[ 1 ] );
+		} else {
+			src = new URL( 'http:' + scriptTagAttributes[ 1 ] );
+		}
+
+		if ( ! src.search ) {
+			setErrorNotice();
+			return;
+		}
+
+		const searchParams = new URLSearchParams( src.search );
+		let styleSetting = searchParams.get( 'theme' );
+		if ( searchParams.get( 'type' ) === 'button' ) {
+			styleSetting = searchParams.get( 'type' );
+		}
+
+		const newAttributes = {
+			rid: searchParams.getAll( 'rid' ),
+			iframe: Boolean( searchParams.get( 'iframe' ) ),
+			domain: searchParams.get( 'domain' ),
+			lang: searchParams.get( 'lang' ),
+			newtab: Boolean( searchParams.get( 'newtab' ) ),
+			style: styleSetting,
+		};
+
+		const validatedNewAttributes = getValidatedAttributes( defaultAttributes, newAttributes );
+		setAttributes( validatedNewAttributes );
+	};
+
+	const styleOptions = getStyleOptions( rid );
+	const styleValues = getStyleValues( rid );
+
+	const updateStyle = newStyle => {
+		setAttributes( { style: newStyle } );
+	};
+
+	const getTypeAndTheme = fromStyle =>
+		rid.length > 1
+			? [ 'multi', 'button' !== fromStyle ? fromStyle : 'standard' ]
+			: [
+					'button' === fromStyle ? 'button' : 'standard',
+					'button' === fromStyle ? 'standard' : fromStyle,
+			  ];
+
+	const blockPreview = styleOveride => {
+		const [ type, theme ] = getTypeAndTheme( styleOveride ? styleOveride : style );
+		return (
+			<>
+				<div className={ `${ className }-overlay` }></div>
+				<iframe
+					title={ `Open Table Preview ${ clientId }` }
```

Comment:
Is this something we'd want to translate?

Reply:
✅